torchgeo/datasets/sentinel.py

```diff
@@ -25,6 +25,221 @@ class Sentinel(RasterDataset):
+    * `Microsoft's Planetary Computer
+      <https://planetarycomputer.microsoft.com/dataset/sentinel-1-rtc>`_
+
+    Product Types:
+
+    * `Level-0
+      <https://sentinels.copernicus.eu/web/sentinel/user-guides/sentinel-1-sar/product-types-processing-levels/level-0>`_:
+      Raw (RAW)
+    * `Level-1
+      <https://sentinels.copernicus.eu/web/sentinel/user-guides/sentinel-1-sar/product-types-processing-levels/level-1>`_:
+      Single Look Complex (SLC)
+    * `Level-1
+      <https://sentinels.copernicus.eu/web/sentinel/user-guides/sentinel-1-sar/product-types-processing-levels/level-1>`_:
+      Ground Range Detected (GRD)
+    * `Level-2
+      <https://sentinels.copernicus.eu/web/sentinel/user-guides/sentinel-1-sar/product-types-processing-levels/level-2>`_:
+      Ocean (OCN)
+
+    Polarizations:
+
+    * HH: horizontal transmit, horizontal receive
+    * HV: horizontal transmit, vertical receive
+    * VV: vertical transmit, vertical receive
+    * VH: vertical transmit, horizontal receive
+
+    Acquisition Modes:
+
+    * `Stripmap (SM)
+      <https://sentinels.copernicus.eu/web/sentinel/user-guides/sentinel-1-sar/acquisition-modes/stripmap>`_
+    * `Interferometric Wide (IW) swath
+      <https://sentinels.copernicus.eu/web/sentinel/user-guides/sentinel-1-sar/acquisition-modes/interferometric-wide-swath>`_
+    * `Extra Wide (EW) swatch
+      <https://sentinels.copernicus.eu/web/sentinel/user-guides/sentinel-1-sar/acquisition-modes/extra-wide-swath>`_
+    * `Wave (WV)
+      <https://sentinels.copernicus.eu/web/sentinel/user-guides/sentinel-1-sar/acquisition-modes/wave>`_
+
+    .. note::
+       At the moment, this dataset only supports the GRD product type. Data must be
+       radiometrically terrain corrected (RTC). This can be done manually using a DEM,
+       or you can download an On Demand RTC product from ASF DAAC.
+
+    .. versionadded:: 0.4
+    """
+
+    # SAFE format
+    # https://sentinels.copernicus.eu/web/sentinel/user-guides/sentinel-1-sar/naming-conventions
+    # https://sentinels.copernicus.eu/web/sentinel/user-guides/sentinel-1-sar/document-library/-/asset_publisher/1dO7RF5fJMbd/content/sentinel-1-product-specification
+
+    # ASF DAAC GRD RTC uses a different naming scheme (README.md.txt in download):
+    #
+    # S1x_yy_aaaaaaaaTbbbbbb_ppo_RTCzz_u_defklm_ssss
+    #
+    # x:          Sentinel-1 Mission (A or B)
+    # yy:         Beam Mode
+    # aaaaaaaa:   Start Date of Acquisition (YYYYMMDD)
+    # bbbbbb:     Start Time of Acquisition (HHMMSS)
+    # pp:         Polarization: Dual-pol (D) vs. Single-pol (S),
+    #                 primary polarization (H vs. V)
+    # o:          Orbit Type: Precise (P), Restituted (R), or Original Predicted (O)
+    # zz:         Terrain Correction Pixel Spacing
+    # u:          Software Package Used: GAMMA (G)
+    # d:          Gamma-0 (g) or Sigma-0 (s) Output
+    # e:          Power (p) or Decibel (d) or Amplitude (a) Output
```

Comment:
The user should specify a scale they want their dataset to be in, and the dataset should normalize all files to that scale. Otherwise, differing images will mean vastly different things.
For reference:
```
amplitude = sqrt(power)
decibel = 10 * log10(power)
decibel = 10 * log10(amplitude ** 2)
```

Reply:
Amplitude data is better for viz, so when plotting you can always plot the sqrt of the power data. I'd recommend normalizing to power by default unless the user wants otherwise.

Reply:
Again, how common is it for a user to have multiple of these formats mixed in a single directory? Right now we just load the data as it's stored in the file. We could add a parameter to `__init__` so that only one of these can be loaded at a single time, or we could modify the scale during loading, but that would require overriding `__getitem__`.